Removing .idea from .gitignore

ankushagarwal opened the pull request:
The project cannot include every editor's
ignore files. Contributers can use a local
global gitignore : https://help.github.com/articles/ignoring-files

ankushagarwal committed:
Removing .idea from .gitignore

The project cannot include every editor's
ignore files. Contributers can use a local
global gitignore : https://help.github.com/articles/ignoring-files

Signed-off-by: Ankush Agarwal <ankushagarwal11@gmail.com>

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "gitignore-iml" git@github.com:ankushagarwal/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

thaJeztah commented:
Hm, I know similar PRs were rejected in the past; you might want to read the comment on line two in this file; https://github.com/docker/docker/pull/14398/files#diff-a084b794bc0759e7a6b77810e01874f2R2

:angel:

thaJeztah commented:
Hm, looking again, it's strange that the `.idea` was accepted?

vdemeester commented:
@thaJeztah Yeah, if ``.idea`` has been accepted ``*.iml`` should too.

But yeah, we can't prevent everything with the ``.gitigignore`` because, Eclipse uses ``.project``, Atom might use something else, etc.. My opinion is that neither ``.idea`` or ``*.iml`` should be there — but being there doesn't hurt either.

thaJeztah commented:
Yes, after looking *when* it was added (waaaay back), and why (no explicit reason, it was a merge from another repository) it may make more sense to remove the `.idea` line instead of adding the `.iml`

ankushagarwal commented:
@thaJeztah @vdemeester : Thanks for the comments. I completely agree. I added .iml after seeing .idea.

Updated PR to delete .idea from the .gitignore file

thaJeztah commented:
Thanks, Ankush

LGTM

ankushagarwal commented:
I am not sure why tests are failing for this change.

vdemeester commented:
@ankushagarwal It's not only your PR, there is a problem with some integration tests, look at https://github.com/docker/docker/pull/14409 :wink:

thaJeztah commented:
@ankushagarwal no worries, it's not related, but something broke the tests, all PRs are failing

thaJeztah commented:
@vdemeester beat me to it :-)

ankushagarwal commented:
Cool. Thanks for fixing the tests @vdemeester.

crosbymichael commented:
LGTM
